packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -171,113 +173,123 @@ class WebAssetServer implements AssetReader {
     bool testMode = false,
     DwdsLauncher dwdsLauncher = Dwds.start,
   }) async {
-    try {
-      InternetAddress address;
-      if (hostname == 'any') {
-        address = InternetAddress.anyIPv4;
-      } else {
-        address = (await InternetAddress.lookup(hostname)).first;
+    InternetAddress address;
+    if (hostname == 'any') {
+      address = InternetAddress.anyIPv4;
+    } else {
+      address = (await InternetAddress.lookup(hostname)).first;
+    }
+    HttpServer httpServer;
+    dynamic lastError;
+    for (int i = 0; i < 5; i += 1) {
```

Comment:
If there's a way to tell that the SocketException is permament I'd do `while (true)`

Reply:
WDYM by permanent?

Reply:
I think generally no, if another program is holding this port we likely can't tell when it will terminate